docs/developer-guide/plugins.md

```diff
@@ -31,6 +33,22 @@ your plugin will respect disabled ranges and other possible future features of s
 - `styleSearch`: Search within CSS strings, and for every match found invoke a callback, passing a match object with details about the match. `styleSearch` ignores CSS strings (e.g. `content: "foo";`) and by default ignores comments. It can also be restricted to substrings within or outside of CSS functional notation.
 - `validateOptions`: Help your user's out by checking that the options they've submitted are valid.
 
+## `stylelint.rules`
+
+All of the rule functions are available at `stylelint.rules`. This allows you to build on top of existing rules for your particular needs.
+
+For example, maybe your codebase uses special comment directives to customize rule options for specific stylesheets. You could build a plugin that checks those directives and then runs the appropriate rules with the right options (or doesn't run them at all).
+
+All rules share a common signature. They are a function that accepts two arguments: a primary option and a secondary options object. And that functions returns a function that has the signature of a PostCSS plugin, expecting a PostCSS root and result as its arguments.
+
+```js
+// Simple illustration of the rule signature
+var colorHexCaseRule = stylelint.rules["color-hex-case"]
+var runColorHexCase = colorHexCaseRule(primaryOption, secondaryOptions)
+runColorHexCase(postcssRoot, postcssResult)
```

Comment:
@davidtheclark This looks great, thanks for adding it! The only thing I could think is to add an `if (..)` around this line, to show one potential use of an existing rule (say, from the plugin). Kinda like what you've done in conditionally-check-color-hex-case. It's really not a big deal though. Or maybe we can just link to src/__tests__/fixtures/plugin-conditionally-check-color-hex-case.js for an example of how it could be used?